GitCommands/Git/GitModule.cs

```diff
@@ -2725,10 +2725,20 @@ public bool IsDirtyDir()
                 : null;
         }
 
+        /// <summary>
+        /// Get the file contents for the HEAD commit
+        /// </summary>
+        /// <param name="file">The Git status item</param>
+        /// <returns>An awaitable task with the requested file contents.</returns>
         public async Task<string?> GetFileContentsAsync(GitItemStatus file)
         {
-            var contents = new StringBuilder();
+            if (!file.IsTracked || (file.Staged == StagedStatus.Index && file.IsNew))
+            {
+                // File is not in Git
+                return null;
```

Comment:
Task is null there.

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/async-return-types

The Task is async, not the same as not async.
The task is not null, it returns a null string.